dicthash/dicthash.py

```diff
@@ -49,58 +45,66 @@ def _save_convert_float_to_int(x):
     return int(x * FLOAT_FACTOR)
 
 
-def _unpack_value(value, prefix='', whitelist=None, blacklist=None):
+def _unpack_value(value: Union[dict, Iterable, float, int],
+                  prefix: str ='', whitelist: Optional[List[Hashable]] =None,
+                  blacklist: Optional[List[Hashable]]=None) -> str:
     """
     Unpack values from a data structure and convert to string. Call
     the corresponding functions for dict or iterables or use simple
     string conversion for scalar variables.
 
     Parameters
     ----------
-    value : dict, iterable, scalar variable
+    value : Union[dict, Iterable, float, int]
         Value to be unpacked.
     prefix : str, optional
         Prefix to preprend to resulting string. Defaults to empty
         string.
+
+    Returns
+    -------
+    str
+        Unpacked values.
     """
 
-    try:
+    if isinstance(value, dict):
         return _generate_string_from_dict(value,
                                           blacklist=blacklist,
                                           whitelist=whitelist,
                                           prefix=prefix + 'd')
-    except AttributeError:
+    else:
         # not a dict
-        try:
+        if isinstance(value, Iterable):
             return prefix + _generate_string_from_iterable(value, prefix='i')
-        except TypeError:
+        else:
             # not an iterable
             if isinstance(value, float):
                 return prefix + str(_save_convert_float_to_int(value))
             else:
                 return prefix + str(value)
 
 
-def _generate_string_from_iterable(l, prefix=''):
+def _generate_string_from_iterable(l: Iterable, prefix: str ='') -> str:
```

Comment:
```suggestion
def _generate_string_from_iterable(l: Iterable, prefix: str='') -> str:
```

Reply:
Style guide says to have whitespace on each sides of the `=`: https://www.python.org/dev/peps/pep-0008/#id53
So I changed in this direction.